### PR TITLE
Simplify gated cfg checking

### PR DESCRIPTION
**Pull request by jseyfried**
r? @nrc

### CODE REVIEW

File: src/librustc_driver/driver.rs [hunk 1]
```diff
@@ -577,15 +577,13 @@ pub fn phase_2_configure_and_expand<'a>(sess: &Session,
...
-                                                     &mut feature_gated_cfgs)
-        })
-    })?;
+    krate = time(time_passes, "configuration", || {
+        let (krate, features) =
+            syntax::config::strip_unconfigured_items(krate, &sess.parse_sess, sess.opts.test);
+        // these need to be set "early" so that expansion sees `quote` if enabled.
+        *sess.features.borrow_mut() = features;
+        krate
+    });
```
Comment (nrc) [inline src/librustc_driver/driver.rs]: Do we need to stop compilation if there were errors here?

Comment (jseyfried) [inline src/librustc_driver/driver.rs]: I don't see any reason to stop compilation -- the only errors that get reported here are malformed `cfg` attributes (in which case the item is kept), malformed `cfg_attr` attributes (in which case the attribute is removed), and malformed `feature` attributes (in which case we might not see the features).
At worst, the user might see some gated feature errors for a feature they intended to use.

Comment (jseyfried) [inline src/librustc_driver/driver.rs]: (also, gated cfg errors are also reported, which definitely don't require us to stop compilation)

**Comment by jseyfried** [comment]
Rebased. @bors r=nrc

**Pull request merged**